clientapi/routing/register.go

```diff
@@ -825,15 +827,25 @@ func handleApplicationServiceRegistration(
 // checkAndCompleteFlow checks if a given registration flow is completed given
 // a set of allowed flows. If so, registration is completed, otherwise a
 // response with
-func checkAndCompleteFlow(
-	flow []authtypes.LoginType,
+func checkAndCompleteFlow(flow []authtypes.LoginType,
 	req *http.Request,
 	r registerRequest,
 	sessionID string,
 	cfg *config.ClientAPI,
 	userAPI userapi.ClientUserAPI,
-) util.JSONResponse {
+	clientRsApi rsAPI.ClientRoomserverAPI) util.JSONResponse {
+
 	if checkFlowCompleted(flow, cfg.Derived.Registration.Flows) {
+		// POST register behavior: add user to room
+		for room := range cfg.AutoJoinRooms {
+			err := addUserToRoom(req.Context(), clientRsApi, cfg.AutoJoinRooms[room], r.Username,
+				userutil.MakeUserID(r.Username, cfg.Matrix.ServerName))
+			if err != nil {
+				log.Fatal(err)
+				return util.JSONResponse{Code: http.StatusInternalServerError,
```

Comment:
Please don't `log.Fatal`, `log.WithError(err).Error("failed to auto-join room")` instead (possibly with information about  which room and user?).
Also I don't think we should fail the registration, only because we couldn't auto-join a room.